dronekit/lib/__init__.py

```diff
@@ -732,7 +782,21 @@ def listener(_, msg):
         @self.on_message('GLOBAL_POSITION_INT')
         def listener(self, name, m):
             (self._lat, self._lon) = (m.lat / 1.0e7, m.lon / 1.0e7)
+            self._relative_alt = m.relative_alt/1000
+            self._location.global_relative_frame = LocationGlobalRelative(self._lat, self._lon, self._relative_alt)
+
             self.notify_attribute_listeners('location', self.location)
+            self.notify_attribute_listeners('location.global_relative_frame', self.location.global_relative_frame)
+            self.location.notify_attribute_listeners('global_relative_frame', self.location.global_relative_frame)
+
+            if self._alt != None or m.alt != 0:
+                # Require first alt value to be non-0
+                # TODO is this the proper check to do?
+                self._alt = m.alt/1000
+                self._location.global_frame = LocationGlobal(self._lat, self._lon, self._alt)
+                self.notify_attribute_listeners('location.global_frame', self.location.global_frame)
+                self.location.notify_attribute_listeners('global_frame', self.location.global_frame)
```

Comment:
Why both notify on vehicle and on location.global frame? FYI, only  ``location.global_frame`` seems to work.